wordpress-mobile/WordPress-Android
# Stats Revamp v2: Pie chart

Fixes #16552 

This adds a pie chart to follower types card and referrers card.

### Notes
- **Referrers Card:** When the app is in English, WordPress and Search pies are shown in the chart. When the app isn't in English, the top two referrers are shown in the pie chart. (Since the group name is localized in the endpoint, we can't find WordPress or Search in referrers endpoint when the app isn't in English. That's why we don't show them in non-English app.)
- **Follower Types Card:** Always 3 follower types (WordPress, Email, Social) are shown.
- If the slice percent is lower than 3.9%, the percent is rounded up to 3.9% to be able to show the slice in the chart without visual corruption.

|Design|PR|
|-|-|
|<img width="320" src="https://user-images.githubusercontent.com/2471769/169718264-ce761da3-38e4-4563-8ae1-04b3e9ca2dc7.png">|<img src="https://user-images.githubusercontent.com/2471769/170681153-ad1691ae-33dd-4bea-8bf2-f41ebab2e835.png" width="320">|


To test:

Setup:
1. Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
2. Select Debug Settings.
3. Find StatsRevampV2FeatureConfig under Features in development enable it and restart app.

Test 1: 
1. Go to stats either using quick links or menu.
2. Ensure that it opens in Insights tab otherwise switch to Insights tab.
3. Notice a new Total Followers card is shown in Insights tab with Line Chart (scroll down if necessary).
4. Tap on "VIEW MORE" on the top right-hand corner of the card.
5. Ensure it opens the detail screen (scroll down if necessary) with Follower Types card.
6. Notice the pie chart view and check its visual data.
7. Ensure WordPress.com Email and Social slices are in the pie chart.

Test 2: 
1. Ensure your device language is English.
2. Go to stats either using quick links or menu.
3. Ensure that it opens in Insights tab otherwise switch to Insights tab.
4. Notice the new Views & Visitors card is shown in Insights tab with Line Chart (scroll down if necessary).
5. Tap on "VIEW MORE" on the top right-hand corner of the card.
6. Notice the pie chart in the referrers card. Ensure its visual is the same as the design.
7. If you use the app in English, ensure `WordPress, Search and Others` legends are in the pie cart.

Test 3:
1. Change the device language to a language different from English.
2. Repeat steps 2-6 in Test 2.
3. Ensure the top two referrers and Others slices are in the pie chart instead of `WordPress, Search and Others`.

## Regression Notes
1. Potential unintended areas of impact
Current referrers card.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Used feature flag to add a pie chart to referrers card and tested manually.

3. What automated tests I added (or what prevented me from doing so)
I didn't add a new test because there is no new feature.

PR submission checklist:
